src/joinRates/JoinRates.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useCallback, useMemo } from 'react'
+
+import { Quality } from '../quality'
+import JoinRatesTable from './JoinRatesTable'
+import { JoinRatesProps } from './model'
+
+const JoinRates = ({
+  data,
+  onRowClickHandler,
+  selectedRules,
+  selectRules,
+  mode,
+  setMode,
+  pagination,
+  smallSize,
+}: JoinRatesProps): JSX.Element => {
+  const [_index, _setIndex] = useState<number>(0)
+  const _joinRatesDqData = useMemo(
+    () => (Array.isArray(data) ? data[_index].dq_result : data.dq_result),
+    [data, _index]
+  )
+  const joinRatesData = useMemo(
+    () =>
+      Array.isArray(data)
+        ? data.map(dataItem => dataItem.dq_join_statistics)
+        : data.dq_join_statistics,
+    [data]
+  )
+
+  const _onRowClickHandler = useCallback(
+    (index: number) => {
+      if (onRowClickHandler) {
+        onRowClickHandler(Array.isArray(data) ? data[index] : data)
+      }
+
+      _setIndex(index)
+    },
+    [onRowClickHandler, _setIndex]
+  )
+
+  return (
+    <>
+      <JoinRatesTable
+        data={joinRatesData}
+        onRowClickHandler={_onRowClickHandler}
+        selectedIndex={_index}
```

Comment:
`selectedIndex` => `selectedJoinRateId`
We need to consider using `id`, not `index`.
We have an `id` for join rates coming from the BE side.

Reply:
here is an example of data - https://github.com/stoys-io/stoys-ui/pull/5/files#diff-0c7ba56574e90fb61465a411b8fa5bba0cd86fab707e29706fdd173f88dd7048 and it didn't have any id